File: recap/ZigZag/ZigZag_lin.py
n = int(raw_input())
a = map(int, raw_input().split())
r = 1
prevdelta = 0
prevval = a[0]
for i in xrange(n):
   if a[i] != prevval:
      delta = a[i] - prevval
      if prevdelta == 0 or delta * prevdelta < 0:
         r += 1
      prevval = a[i]
      prevdelta = delta
print r

File: recap/ZigZag/ZigZag_lin.cpp
#include <iostream>
using namespace std ;
int main() {
   int n, v, r=1, prevdelta=0, prevval ;
   cin >> n >> prevval ;
   for (int i=1; i<n; i++) {
      cin >> v ;
      if (v != prevval) {
         int delta = v - prevval ;
         if (prevdelta == 0 || delta * prevdelta < 0)
            r++ ;
         prevval = v ;
         prevdelta = delta ;
      }
   }
   cout << r << endl ;
}
